log-collector-script/linux/eks-log-collector.sh

```diff
@@ -221,6 +221,14 @@ get_region() {
   fi
 }
 
+get_user_data() {
+  if USER_DATA=$(curl -H "X-aws-ec2-metadata-token: $IMDS_TOKEN" -f -s --max-time 10 --retry 5 http://169.254.169.254/latest/user-data); then
+    echo "${USER_DATA}" > "${COLLECT_DIR}"/system/user-data.txt
+  else
+    warning "Unable to find EC2 user-data, skipping."
+  fi
+}
```

Comment:
I'm not sure if it is safe to `echo` without handling special case?

I think it would be better with `base64 / base64 -d`, WDYT?

Test results:

```bash
[root@ip-XXX ~]# rm -vf user-data.txt
removed ‘user-data.txt’
```

```bash
[root@ip-XXX ~]# cat test.sh
#!/usr/bin/env bash

IMDS_TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 360")

if USER_DATA_BASE64=$(curl -H "X-aws-ec2-metadata-token: $IMDS_TOKEN" -f -s --max-time 10 --retry 5 http://169.254.169.254/latest/user-data | base64); then
  echo "${USER_DATA_BASE64}" | base64 -d > user-data.txt
else
  # changed "warning" to "echo" for test
  echo "Unable to find EC2 user-data, skipping."
fi
```

```bash
[root@ip-XXX ~]# ./test.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    56  100    56    0     0   6820      0 --:--:-- --:--:-- --:--:--  7000
```

```bash
[root@ip-XXX ~]# cat user-data.txt
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="//"

... (omit)
```